### ISSUE TITLE
Add learning rate scheduler as hyperparameter

### ISSUE THREAD

**Issue by edwardchalstrey1** [alan-turing-institute/autoemulate#576]
### Summary

Update PytorchBackend and GPytorchBackend to include a learning rate scheduler, see [PyTorch docs](https://docs.pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate)

- This would be an additional model config parameter
- May want to also include other params e.g. gamma
    - Things like which scheduler, optimizer to use (see momentum param below used by SGD but not adam)
- Note: Look at the MLP emulator in experimental where these params are already passed
- **Look into passing in the scheduler as an argument, instead of the individual hyperparameters** (partially instantiated e.g. with a default learning rate set)

### Basic Example

PyTorch example:
```
optimizer = optim.SGD(model.parameters(), lr=0.01, momentum=0.9)
scheduler = ExponentialLR(optimizer, gamma=0.9)

for epoch in range(20):
    for input, target in dataset:
        optimizer.zero_grad()
        output = model(input)
        loss = loss_fn(output, target)
        loss.backward()
        optimizer.step()
    scheduler.step()
```

### TODO:

- [x] Add optimizer as an argument to `PyTorchBackend` with a default
- [x] Add learning rate as an argument to `PyTorchBackend` with a default
- [x] Add scheduler argument to `PyTorchBackend`
- [x] Add scheduler in training loop
- [x] Add gamma as an argument to `PyTorchBackend` with a default

### Drawbacks

None, this would be done as part of 1.0 release instead of #475
